src/ocamlorg_web/lib/handler.ml

```diff
@@ -227,7 +227,11 @@ let carbon_footprint = page Url.carbon_footprint
 let privacy_policy = page Url.privacy_policy
 let governance = page Url.governance
 let code_of_conduct = page Url.code_of_conduct
-let playground _req = Dream.html (Ocamlorg_frontend.playground ())
+
+let playground _req =
+  let default = Data.Code_example.get "default.ml" in
+  let default_code = Omd.escape_html_entities default.body in
```

Comment:
I was worried for a moment in the template when I saw `<%s!` (which means "render this unescaped"), but then I see it's being escaped here.

I would recommend moving the HTML escaping to the template and use `Dream.html_escape` instead of `omd`'s facilities. Then, the handler deals with data and the template deals with rendering and escaping.

Generally, this splitting of responsibilities is pretty powerful in terms of the flexibility it gives us regarding rendering - but it is currently being violated in some places because ood-gen is rendering HTML snippets in advance. In the case of the table of contents of tutorials https://github.com/ocaml/ocaml.org/pull/1106), I had to undo the HTML-snippet rendering in `ood-gen` in order to get at the underlying data.

Reply:
I was wondering about the `<%s!` vs `<%s` tags.
It sounds better to do the escaping in the template indeed.

Reply:
Oki great, this stuff isn't all self-documenting, so I am glad it was useful!